students/jeff_shabani/session07/html_render.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element(object):
+
+    def __init__(self, content=None):
```

Comment:
Hi Jeff,

For this one, as I think Michael mentioned yesterday, if you force _None_ into a string, it will turn into the string "None". The same would apply to boolean values, for example. One way to do it, without changing the default value for _content_, would be:

```
    def __init__(self, content=None):
        if content:
            self.content = content
        else:
            self.content = '' # Assigns an empty string
```

Note that I'm not using _str()_ to force _content_ into a string. In this case, you should assume _content_ will always be either _None_ or a string to begin with. Your code should crash (or raise an exception) if you get something else (like an integer). This is important when you consider the fact that you will eventually be writing Python modules that other people will be using, situations in which the user enters the wrong kind of input should trigger an error (or at least a warning) before you try to force that input into the right kind (for example, by converting it to a string).

-Luis

Reply:
Thanks Luis. I realized that class methods are really just functions. It's the knowing when and when not to use self. Also, great office hours last night.